builtin/credential/aws/path_login.go

```diff
@@ -1467,11 +1468,15 @@ func parseIamRequestHeaders(headersB64 string) (http.Header, error) {
 		switch typedValue := v.(type) {
 		case string:
 			headers.Add(k, typedValue)
+		case json.Number:
+			headers.Add(k, typedValue.String())
 		case []interface{}:
 			for _, individualVal := range typedValue {
 				switch possibleStrVal := individualVal.(type) {
 				case string:
 					headers.Add(k, possibleStrVal)
+				case json.Number:
+					headers.Add(k, possibleStrVal.String())
 				default:
 					return nil, fmt.Errorf("header %q contains value %q that has type %s, not string", k, individualVal, reflect.TypeOf(individualVal))
```

Comment:
This error message and the one a few lines below are now incorrect since it also accepts json.Number.

Reply:
@joelthompson I actually thought about that and was about 60% leaving it this way and 40% changing it. The reason is twofold: one, technically json.Number is a special casing of a string inside Go with some extra parsing logic; the other is that headers are supposed to always be strings, so we are now *accepting* ints for content-length but it's still incorrect.